src/messaging.rs

```diff
@@ -218,4 +215,10 @@ impl<NodeUid: Clone + Ord> NetworkInfo<NodeUid> {
     pub fn invocation_id(&self) -> Vec<u8> {
         self.public_key_set.public_key().to_bytes()
     }
+
+    /// Returns `true` if this node takes part in the consensus itself. If not, it is only an
+    /// observer.
+    pub fn is_full_node(&self) -> bool {
```

Comment:
Aha, by "full" you mean "not a listener". To me the name alludes to size. Maybe it can be called `is_peer` since we have two kinds of node: peer and listener?

Reply:
I renamed it to `is_peer` but kept the word "observer" for now — do you prefer "listener" for that?

Reply:
Thanks! "Observer" is perfect, I just used a wrong term.